node_cli/utils/meta.py

```diff
@@ -11,62 +13,141 @@
+            'version': self.version,
+            'config_stream': self.config_stream,
+            'os_id': self.os_id,
+            'os_version': self.os_version,
+        }
+
 
+class BaseCliMetaManager(abc.ABC):
+    def __init__(self, meta_filepath: str = META_FILEPATH) -> None:
+        self.meta_filepath = meta_filepath
 
-def get_meta_info(raw: bool = False) -> CliMeta:
-    if not os.path.isfile(META_FILEPATH):
-        return None
-    with open(META_FILEPATH) as meta_file:
-        plain_meta = json.load(meta_file)
-    if raw:
-        return plain_meta
-    return CliMeta(**plain_meta)
+    def _get_plain_meta(self) -> dict:
+        if not os.path.isfile(self.meta_filepath):
+            return {}
+        with open(self.meta_filepath) as meta_file:
+            return json.load(meta_file)
 
+    @abc.abstractmethod
+    def get_meta_info(self, raw: bool = False) -> CliMetaBase | dict | None:
+        pass
 
-def save_meta(meta: CliMeta) -> None:
-    with open(META_FILEPATH, 'w') as meta_file:
-        json.dump(meta._asdict(), meta_file)
+    def save_meta(self, meta: CliMetaBase) -> None:
+        with open(self.meta_filepath, 'w') as meta_file:
+            json.dump(meta.asdict(), meta_file)
 
+    @abc.abstractmethod
+    def compose_default_meta(self) -> CliMetaBase:
+        pass
 
-def compose_default_meta() -> CliMeta:
-    return CliMeta(
-        version=DEFAULT_VERSION,
-        docker_lvmpy_stream=DEFAULT_DOCKER_LVMPY_STREAM,
-        config_stream=DEFAULT_CONFIG_STREAM,
-        os_id=DEFAULT_OS_ID,
-        os_version=DEFAULT_OS_VERSION,
-    )
+    def ensure_meta(self, meta: CliMetaBase | None = None) -> None:
+        if not self.get_meta_info():
+            meta = meta or self.compose_default_meta()
+            self.save_meta(meta)
 
+    @abc.abstractmethod
+    def update_meta(self, *args, **kwargs) -> None:
+        pass
 
-def ensure_meta(meta: Optional[CliMeta] = None) -> None:
-    if not get_meta_info():
-        meta = meta or compose_default_meta()
-        save_meta(meta)
 
+class CliMetaManager(BaseCliMetaManager):
+    def get_meta_info(self, raw: bool = False) -> CliMeta | dict | None:
+        plain_meta = self._get_plain_meta()
+        if not raw and not plain_meta:
+            return None
+        allowed_fields = set(CliMeta.__dataclass_fields__.keys())
+        clean_plain_meta = {k: v for k, v in plain_meta.items() if k in allowed_fields}
+
+        if raw:
+            return clean_plain_meta
+        return CliMeta(**clean_plain_meta)
+
+    def compose_default_meta(self) -> CliMeta:
+        return CliMeta(
+            version=DEFAULT_VERSION,
+            docker_lvmpy_stream=DEFAULT_DOCKER_LVMPY_STREAM,
+            config_stream=DEFAULT_CONFIG_STREAM,
+            os_id=DEFAULT_OS_ID,
+            os_version=DEFAULT_OS_VERSION,
+        )
+
+    def update_meta(
+        self,
+        version: str,
+        config_stream: str,
+        docker_lvmpy_stream: Optional[str],
```

Comment:
should be marked as optional in the parent class or marked as required in this class:

```
Argument of type "str | None" cannot be assigned to parameter "docker_lvmpy_stream" of type "str" in function "__init__"
  Type "str | None" is not assignable to type "str"
    "None" is not assignable to "str"Pylance[reportArgumentType](https://github.com/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md)
(parameter) docker_lvmpy_stream: str | None
```